tools/diff-processor/diff/diff.go

```diff
@@ -0,0 +1,231 @@
+package diff
+
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// SchemaDiff is a nested map with field names as keys and Field objects
+// as bottom-level values.
+// Fields are assumed not to be covered until detected in a test.
+type SchemaDiff map[string]ResourceDiff
+
+type ResourceDiff struct {
+	ResourceConfig ResourceConfigDiff
+	Fields map[string]FieldDiff
+}
+
+type ResourceConfigDiff struct {
+	Old *schema.Resource
+	New *schema.Resource
+}
+
+type FieldDiff struct {
+	Old *schema.Schema
+	New *schema.Schema
+}
+
+func ComputeSchemaDiff(oldResourceMap, newResourceMap map[string]*schema.Resource) SchemaDiff {
```

Comment:
This function is quite large. Could you break this up into sub functions so it's more readable ?

Reply:
I've factored out the key union logic - I'm not sure what else I would be able to factor out, and it's only ~30 lines at this point, and it's all closely related. Let me know if there's anything specific you think would make sense to factor out.